src/main.cpp

```diff
@@ -7622,70 +7622,72 @@ bool SendMessages(CNode *pto)
             {
                 fSendTrickle = true;
                 pto->nNextInvSend = PoissonNextSend(nNow, AVG_INVENTORY_BROADCAST_INTERVAL);
             }
-            LOCK(pto->cs_inventory);
-            vInv.reserve(std::min<size_t>(1000, pto->vInventoryToSend.size()));
-            vInvWait.reserve(pto->vInventoryToSend.size());
-            BOOST_FOREACH (const CInv &inv, pto->vInventoryToSend)
+
+            if (1)
             {
-                if (inv.type == MSG_TX && pto->filterInventoryKnown.contains(inv.hash))
-                    continue;
+                // BU - here we only want to forward message inventory if our peer has actually been requesting
+                // useful data or giving us useful data.  We give them 2 minutes to be useful but then choke off
+                // their inventory.  This prevents fake peers from connecting and listening to our inventory
+                // while providing no value to the network.
+                // However we will still send them block inventory in the case they are a pruned node or wallet
+                // waiting for block announcements, therefore we have to check each inv in pto->vInventoryToSend.
+                bool chokeTxInv = (pto->nActivityBytes == 0 && (nNow / 1000000 - pto->nTimeConnected) > 120);
+                LOCK(pto->cs_inventory);
 
-                // BU - here we only want to forward message inventory if our peer has actually been requesting useful
-                // data or
-                //      giving us useful data.  We give them 2 minutes to be useful but then choke off their inventory.
-                //      This
-                //      prevents fake peers from connecting and listening to our inventory while providing no value to
-                //      the network.
-                //      However we will still send them block inventory in the case they are a pruned node or wallet
-                //      waiting for block
-                //      announcements, therefore we have to check each inv in pto->vInventoryToSend.
-                if (inv.type == MSG_TX && pto->nActivityBytes == 0 && (GetTime() - pto->nTimeConnected) > 120)
-                {
-                    LogPrint("evict", "choking off tx inventory for %s time connected %d\n", pto->addr.ToString(),
-                        GetTime() - pto->nTimeConnected);
-                    continue;
-                }
+                vInvSend.reserve(pto->vInventoryToSend.size());
+                // about 1/4 of the nodes should end up in this list, so over-allocate
+                vInvWait.reserve(pto->vInventoryToSend.size() / 3);
 
-                // trickle out tx inv to protect privacy
-                if (inv.type == MSG_TX && !fSendTrickle)
+                // Make copy of vInventoryToSend while cs_inventory is locked but also ignore some tx and defer others
+                BOOST_FOREACH (const CInv &inv, pto->vInventoryToSend)
                 {
```

Comment:
why not using c++11 for loop ?

Reply:
Its the original code, just indented differently... I'll change to c++11

Reply:
great